xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/resources/notification-filter-preferences.hbm.xml

```diff
@@ -48,19 +50,23 @@
 
     <property name="startingDate" type="timestamp" column="nfp_date" />
 
-    <property name="allEventTypes" type="text" column="nfp_event_types" />
-
-    <property name="user" type="string" column="nfp_user" length="768"/>
-
-    <property name="pageOnly" type="string" column="nfp_page_only" length="768" />
+    <set name="eventTypes"  table="notification_filter_prefs_eventTypes" lazy="false">
+      <key column="nfp_id" />
+      <element column="value" type="string" length="768" />
+    </set>
 
-    <property name="page" type="string" column="nfp_page" length="768" />
-
-    <property name="wiki" type="string" column="nfp_wiki" length="255" />
+    <property name="scope" column="nfp_scope" not-null="true">
+      <type name="org.hibernate.type.EnumType">
+        <param name="enumClass">org.xwiki.notifications.filters.NotificationFilterScope</param>
```

Comment:
Note that by default, hibernate will store the enum element index. It's most probably a slightly better for performances, but it's something I personally find quite fragile (you break everything if you change in any way the order of the enum element or insert a new element anywhere else than at the end).

To store the enum name instead, you can use https://github.com/xwiki-contrib/application-replication/blob/master/replication-entity/replication-entity-controller/src/main/resources/replication/entityreplication.hbm.xml#L40.

Reply:
Hmmm so maybe something to change also for the filterType while I'm at it.